paasta_tools/prune_completed_pods.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import logging
+import socket
+import sys
+from datetime import datetime
+from typing import Sequence
+
+from dateutil.tz import tzutc
+from kubernetes.client import V1Pod
+
+from paasta_tools.kubernetes_tools import get_all_pods
+from paasta_tools.kubernetes_tools import get_pod_condition
+from paasta_tools.kubernetes_tools import is_pod_completed
+from paasta_tools.kubernetes_tools import KubeClient
+
+log = logging.getLogger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Terminates completed pods based on namespace and minutes since completion"
+    )
+    parser.add_argument(
+        "-n",
+        "--namespace",
+        help="Namespace of the pods to terminate from",
+        required=True,
+    )
+    parser.add_argument(
+        "-m",
+        "--minutes",
+        default=0,
+        help="Minutes since the pods' completion. Terminates pods based on time since completion.",
+        required=True,
+    )
+    parser.add_argument(
+        "--dry-run",
+        dest="dry_run",
+        action="store_true",
+        help="Print pods to be terminated, instead of terminating them",
+    )
+    parser.add_argument(
+        "-v", "--verbose", dest="verbose", action="store_true", default=False
+    )
+    args = parser.parse_args()
+    return args
+
+
+def setup_logging(verbose):
+    level = logging.DEBUG if verbose else logging.WARNING
+    logging.basicConfig(level=level)
+
+
+def completed_since(pod: V1Pod, allowed_uptime_minutes: int) -> bool:
+    seconds_per_minute = 60
+    time_finished = get_pod_condition(pod, "ContainersReady").last_transition_time
+    time_now = datetime.now(tzutc())
+    # convert total seconds since completion to minutes
+    completed_since = (time_now - time_finished).total_seconds() / seconds_per_minute
+
+    return completed_since > allowed_uptime_minutes
+
+
+def terminate_pods(pods: Sequence[V1Pod], kube_client) -> tuple:
+    successes = []
+    errors = []
+
+    for pod in pods:
+        try:
+            kube_client.core.delete_namespaced_pod(
+                name=pod.metadata.name,
+                namespace=pod.metadata.namespace,
+                grace_period_seconds=0,
+                propagation_policy="Background",
+            )
+            successes.append(pod.metadata.name)
+        except Exception as e:
+            errors.append((pod.metadata.name, e))
+
+    return (successes, errors)
+
+
+def main():
+    args = parse_args()
+
+    kube_client = KubeClient()
+    pods = get_all_pods(kube_client, args.namespace)
+    allowed_uptime_minutes = int(args.minutes)
+    completed_pods = []
+
+    setup_logging(args.verbose)
+
+    for pod in pods:
+        if is_pod_completed(pod) and completed_since(pod, allowed_uptime_minutes):
+            completed_pods.append(pod)
+
+    if len(completed_pods) < 1:
```

Comment:
```suggestion
    if not completed_pods:
```
might be a little more idiomatic - but feel free to ignore me

Reply:
Done 👍